matchiter.go

```diff
@@ -35,7 +35,8 @@ type candidateMatch struct {
 	byteOffset  uint32
 	byteMatchSz uint32
 
-	symbolInfo *Symbol
+	symbol    bool
+	symbolIdx uint32
```

Comment:
we can do better with alignment and save space. Check the proposed place to put these fields, just after the bools at the top of the strucT: http://golang-sizeof.tips/?t=c3RydWN0IHsKCWNhc2VTZW5zaXRpdmUgYm9vbAoJZmlsZU5hbWUgICAgICBib29sCgogICAgc3ltYm9sICAgIGJvb2wKICAgIHN5bWJvbElkeCB1aW50MzIKCglzdWJzdHJCeXRlcyAgIFtdYnl0ZQoJc3Vic3RyTG93ZXJlZCBbXWJ5dGUKCglmaWxlIHVpbnQzMgoKCS8vIE9mZnNldHMgYXJlIHJlbGF0aXZlIHRvIHRoZSBzdGFydCBvZiB0aGUgZmlsZW5hbWUgb3IgZmlsZSBjb250ZW50cy4KCXJ1bmVPZmZzZXQgIHVpbnQzMgoJYnl0ZU9mZnNldCAgdWludDMyCglieXRlTWF0Y2hTeiB1aW50MzIKCn0K

Reply:
Interesting, I didn't know this was the case. I figure this would be an optimization that the Go compiler handles.